### PR TITLE
feat: added cli support

### PR DESCRIPTION
**Pull request by prashant-15**
added cli support along with host, port flags

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -18,11 +18,11 @@ func getRedisConfig(cfgFile string) *RedisConfig {
 	return &config
 }
 
-func newRedisClient(cfgFile string) *redis.Client {
+func newRedisClient(cfgFile string, Addr string) *redis.Client {
 	config := getRedisConfig(cfgFile)
 	if config.ServerConfigs != nil {
 		return redis.NewClient(&redis.Options{
-			Addr:     config.ServerConfigs.Address,
+			Addr: Addr,
```
Comment (darkLord19) [inline client.go]: use from cli if passed else fallback on config

Comment (97amarnathk) [reply client.go]: @darkLord19 isse better caller can decide what to pass. 
This function can then be taken to a new package

File: main.go [hunk 2]
```diff
@@ -37,39 +42,66 @@ func getMatchingKeys(pattern string, matchedKeys chan []string, client *redis.Cl
...
-	argsWithoutProg := os.Args[1:]
-	lenghtOfArgsWithoutProg := len(argsWithoutProg)
-	if lenghtOfArgsWithoutProg == 0 {
-		log.Fatalln("No key patterns provided")
-	}
+	app := &cli.App{
+		Name:  "redisdel",
+		Usage: "Scan for redis keys matching a given pattern and delete them",
+		Version: "0.1.0",
+		Action: func(cCtx *cli.Context) error {
```
Comment (97amarnathk) [inline main.go]: possible to extract action to another function in the package ? would be easier to read

File: go.mod [hunk 3]
```diff
@@ -9,6 +9,10 @@ require (
 
 require (
 	github.com/cespare/xxhash/v2 v2.1.2 // indirect
+	github.com/cpuguy83/go-md2man/v2 v2.0.2 // indirect
 	github.com/dgryski/go-rendezvous v0.0.0-20200823014737-9f7001d12a5f // indirect
+	github.com/russross/blackfriday/v2 v2.1.0 // indirect
+	github.com/urfave/cli/v2 v2.23.5 // indirect
```
Comment (97amarnathk) [inline go.mod]: might need to run `go mod tidy` as `cli` is a direct dependency

File: client.go [hunk 4]
```diff
@@ -18,11 +18,11 @@ func getRedisConfig(cfgFile string) *RedisConfig {
 	return &config
 }
 
-func newRedisClient(cfgFile string) *redis.Client {
+func newRedisClient(cfgFile string, Addr string) *redis.Client {
```
Comment (97amarnathk) [inline client.go]: [optional] can make an interface for this as we will have many options.

File: client.go [hunk 5]
```diff
@@ -18,11 +18,11 @@ func getRedisConfig(cfgFile string) *RedisConfig {
...
 	config := getRedisConfig(cfgFile)
 	if config.ServerConfigs != nil {
 		return redis.NewClient(&redis.Options{
-			Addr:     config.ServerConfigs.Address,
+			Addr: Addr,
 			Password: config.Password,
```
Comment (97amarnathk) [inline client.go:11]: take Password, Username from caller too 
(caller can decide where this value is derived from cli or config)

**Review by 97amarnathk — APPROVED** [review]

File: client.go (hunk 1 above)
Comment (darkLord19) [reply client.go]: works. newRedicClient ke input me can take model.RedisConfig as input

File: client.go (hunk 4 above)
Comment (darkLord19) [reply client.go]: it is already there. RedisConfig

File: main.go [hunk 6]
```diff
@@ -2,48 +2,63 @@ package main
...
-	client := newRedisClient("redisdel.conf")
+	addr := fmt.Sprintf("%s:%s", host, port)
+	fileConfig := getRedisConfig("redisdel.conf")
+	redisServerConfig := RedisServerConfigs{
+		Address: addr,
+	}
+	if redisServerConfig.Address == "" {
+		redisServerConfig.Address = fileConfig.ServerConfigs.Address
+	}
+
```
Comment (darkLord19) [inline main.go]: create a new file named cli.go or cmd.go and move this in there

**Review by darkLord19 — APPROVED** [review]

**Comment by darkLord19** [comment]
can merge after resolving merge conflict

**Pull request merged**